BitmapEngine/src/main.c

```diff
@@ -3,75 +3,81 @@
 /**
  * Runs formatter/compressor/query engines as set in Control.h
  *
- *
- * format:		F	BITMAP_FILE
- * compress:	C 	BITMAP_FILE		NUM_THREADS		STRIPED/UNSTRIPED
- * query: 		Q 	BITMAP_PATH		QUERY_FILE		NUM_THREADS
- *
+ * format:      F   bitmap_file
+ * compress:    C   bitmap_file     striped/unstriped   num_threads
+ * query:       Q   bitmap_path     query_file          num_threads
  */
 int main(int argc, char *argv[])
 {
-
     setbuf(stdout, NULL);
-    // TODO: this function is overloaded. I think rather than using flags to tell the function what to do,
-    // move each function to a separate one to simplify the logic of this code.
-    // e.g., bformat, bcompress, bquery
 
-    if (argc > 2 && (strcmp(argv[1], "F") == 0 || strcmp(argv[1], "C") == 0 || strcmp(argv[1], "Q") == 0))
+    /* FORMAT */
+    if (argc == 3 && strcmp(argv[1], "F") == 0)
     {
+        char* bitmap_file = argv[2];
 
-        if (argc == 3 && strcmp(argv[1], "F") == 0) // FORMATTING
+        int reformatted = reformat( & bitmap_file);
+
+        if (reformatted == 0)
         {
-            printf("formatting recognized\n");
-            if (reformat( & argv[2]) == 0)
-            {
-                printf("Unsuccessful reformatting of %s\n", argv[2]);
-            }
+            printf("Unsuccessful reformatting of %s\n", bitmap_file);
+        }
+    }
+
+    /* COMPRESS */
+    else if (argc == 5 && strcmp(argv[1], "C") == 0)
+    {
+        int num_threads = atoi(argv[4]);
+        if (num_threads < 1) return -1;
+
+        char* format = argv[3];
+        int striped = strcmp(format, "STRIPED");
+        int unstriped = strcmp(format, "UNSTRIPED");
+
+        if (striped != 0 && unstriped != 0) return -1;
+
+        char* bitmap_file = argv[2];
 
+        char results_name[BUFF_SIZE];
+        /* where the results are being stored */
+        snprintf(results_name, BUFF_SIZE, "%s_RESULTS.csv", bitmap_file);
+
+        double time;
+
+        /* run compression here */
+        if (strcmp(format, "UNSTRIPED") == 0)
+        {
+            time = compress(bitmap_file, UNSTRIPED, BBC, n);

```

Comment:
When I compile this I get an error saying `n` is undeclared. Please test your code before submitting a review.

Reply:
674b0d7